src/main/java/net/minestom/server/network/ComponentNetworkBufferTypeImpl.java

```diff
@@ -247,7 +254,28 @@ private void writeHoverEvent(@NotNull NetworkBuffer buffer, @NotNull HoverEvent<
 
             buffer.write(BYTE, TAG_COMPOUND);
             writeUtf(buffer, "components");
-            //todo item components
+            BinaryTagWriter nbtWriter = impl(buffer).nbtWriter();
+            BinaryTagSerializer.Context context = impl(buffer).registries() == null ? BinaryTagSerializer.Context.EMPTY : new BinaryTagSerializer.ContextWithRegistries(impl(buffer).registries(), true);
+            for (var entry : value.dataComponents().entrySet()) {
+                if (entry.getValue() instanceof MinestomDataComponentValue minestomDataComponentValue && minestomDataComponentValue.component().isSerialized()) {
+                    // If it's a removed value
+                    if (minestomDataComponentValue.value().equals(DataComponentValue.removed())) {
```

Comment:
this is wrong, it should be a check on `entry.getValue()`, not the value of our MinestomDataComponentValue which only holds added components.